File: LeetcodeQuestions/leetcode_challenges/February_challange/containerWithMostWater.js
var maxArea = function (height) {
    let maxArea = -Infinity;

    let i = 0;
    let j = height.length - 1;

    while (i < j) {
        let currentArea = Math.min(height[i], height[j]) * (j - i)
        maxArea = Math.max(maxArea, currentArea)
        if (height[i] < height[j]) {
            i++
        } else {
            j--;
        }
    }
    return maxArea;
};

console.log(maxArea([1, 8, 6, 2, 5, 4, 8, 3, 7]))
console.log(maxArea([2, 3, 4, 5, 18, 17, 6]))

File: LeetcodeQuestions/leetcode_challenges/February_challange/container_with_most_water.rb
def max_area(height)
    max_area = -Float::INFINITY
    i = 0
    j = height.length - 1
    
    while i < j
        current_area = [height[i], height[j]].min * (j - i)
        max_area = [max_area, current_area].max
        if (height[i] < height[j]) 
            i += 1
        else
            j -= 1
        end
    end
    max_area
end